mirobo/fan.py

```diff
@@ -0,0 +1,192 @@
+import logging
+from .device import Device
+from typing import Any, Dict
+import enum
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class LedBrightness(enum.Enum):
+    Bright = 0
+    Dim = 1
+    Off = 2
+
+
+class MoveDirection(enum.Enum):
+    Left = 'left'
+    Right = 'right'
+
+
+class FanStatus:
+    """Container for status reports from the Xiaomi Smart Fan."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        # ['temp_dec', 'humidity', 'angle', 'speed', 'poweroff_time', 'power',
+        # 'ac_power', 'battery', 'angle_enable', 'speed_level',
+        # 'natural_level', 'child_lock', 'buzzer', 'led_b', 'led']
+        #
+        # [232, 46, 30, 298, 0, 'on', 'off', 98, 'off', 1, 0, 'off', 'on',
+        # 1, 'on']
+        self.data = data
+
+    @property
+    def power(self) -> str:
+        return self.data["power"]
+
+    @property
+    def is_on(self) -> bool:
+        return self.power == "on"
+
+    @property
+    def humidity(self) -> int:
+        return self.data["humidity"]
+
+    @property
+    def temperature(self) -> float:
+        if self.data["temp_dec"] is not None:
+            return self.data["temp_dec"] / 10.0
+        else:
+            return None
```

Comment:
I think it's better to skip the `else` block completely and just return False if the `if` doesn't get hit.